parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -173,7 +182,8 @@ write_files:
   content: !!binary |
     {{GetVariableProperty "cloudInitData" "kubeletMonitorSystemdService"}}
 
-{{- if NeedsContainerd}}
+{{- if not IsKrustlet}}
```

Comment:
I did some testing on this and unfortunately it doesn't work super cleanly. Can we reenable containerd and docker for now?

We have a ton of container images prebaked into the VHD. At the end of bootstrapping, we use docker/containerd and a corresponding CLI to delete all of them (we're talking gigabytes of data here, very impactful, but after bootstrap so it doesn't affect provisioning latency).

With krustlet, if we disable containerd/docker, we can't do that cleanup. This would be one reason for example having a separate VHD might have been nice. I think for now, we can just let containerd be installed so we can do cleanup properly, and down the road we can iterate.

Reply:
https://github.com/Azure/AgentBaker/blob/a59bc79933e073baa740a55a7f3b006438bcaf5d/parts/linux/cloud-init/artifacts/cse_main.sh#L75 is the cleanup i'm talking about as reference.